benchmarks/float8/bench_grouped_mm.py

```diff
@@ -0,0 +1,222 @@
+
+    for idx, (name, (M, K, N, E)) in enumerate(
+        name_to_shapes,
+    ):
+        if n_limit is not None and idx >= n_limit:
+            break
+        assert M % E == 0, (
+            "tokens (M) must be evenly divisible by num experts (E) for this benchmark"
+        )
+        tops = 2 * M * N * K * E
+        print("M, K, N, E:", M, K, N, E, f"tops: {tops:.2E}")
+
+        # Run bf16 torch._grouped_mm baseline.
+        A = torch.randn(M, K, device=device, dtype=dtype)
+        B = torch.randn(E, K, N, device=device, dtype=dtype)
+        group_size = M // E
+        offs = torch.arange(
+            group_size, M + 1, group_size, dtype=torch.int32, device=device
+        )
+        ref_time_sec, ref_tops_sec, ref_pct_top_peak = do_benchmarks(
+            tops,
+            bf16_peak_tops,
+            use_gpu_kernel_time,
+            torch._grouped_mm,
+            A,
+            B,
+            offs,
+        )
+        print(
+            f"{dtype} time_sec {ref_time_sec:.2E}, tops/sec {ref_tops_sec:.2E}, pct_peak {ref_pct_top_peak:.3f}"
+        )
+        del A
+        del B
+
+        # Run scaled_grouped_mm.
+        A_hp = torch.randn(M, K, device=device)
+        B_hp_t = (
+            torch.randn(E, K, N, device=device)
+            .transpose(-2, -1)
+            .contiguous()
+            .transpose(-2, -1)
+        )
+
+        if recipe == "rowwise":
+            # TODO: add e5m2
+            A = A_hp.to(torch.float8_e4m3fn)
+            B = B_hp_t.to(torch.float8_e4m3fn)
+            peak_tops = fp8_peak_tops
+            scale_a = torch.ones(M, device=device)
+            scale_b = torch.ones(E, N, device=device)
+        else:
+            assert False, f"unknown recipe {recipe}"
+
+        def do_scaled_grouped_mm(A, B):
+            nonlocal scale_a
+            nonlocal scale_b
+            nonlocal offs
+            return torch._scaled_grouped_mm(A, B, scale_a, scale_b, offs=offs)
+
+        if recipe == "rowwise":
+            do_matmul = do_scaled_grouped_mm
+        else:
+            raise ValueError(f"unknown recipe {recipe}")
+
+        time_sec, tops_sec, pct_top_peak = do_benchmarks(
+            tops, peak_tops, use_gpu_kernel_time, do_matmul, A, B
+        )
+        print(
+            f"time_sec {time_sec:.2E}, tops/sec {tops_sec:.2E}, pct_peak {pct_top_peak:.3f}"
+        )
+
+        del A, B
+        if scale_a is not None:
+            del scale_a
+        if scale_b is not None:
+            del scale_b
+
+        results.append(
+            [
+                name,
+                M,
+                K,
+                N,
+                E,
+                ref_time_sec,
+                time_sec,
+                ref_time_sec / time_sec,
+            ]
+        )
+
+    data_df = pd.DataFrame(results, columns=headers)
+    print(data_df)
+
+    if out_filename is not None:
+        data_df.to_csv(out_filename)
+
+
+def get_name_to_shapes_iter(
```

Comment:
nit: put next to the dense one (I think that one is in utils somewhere), and align the names?